src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -42,6 +42,45 @@
  * &lt;module name=&quot;FinalClass&quot;/&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * final class MyClass {
+ *   private MyClass(){ } //OK
+ * }
+ *
+ * class MyClass {
+ *   private MyClass() { }
+ * } //violation,class should be declared final
+ *
+ * class MyClass {
+ *   int a;String str;
+ *   private MyClass(int a){
+ *   this.a=a;
+ *   str="dummy value";
+ *   }
+ *   public MyClass(String str){
+ *   this.str=str;
+ *   a=0;
+ *   } //OK,since it has a public constructor
```

Comment:
Same, use whitespaces to make your code clear to see.

Reply:
Done

Reply:
Resolved